### PR TITLE
Multistage docker image

### PR DESCRIPTION
**Pull request by jkfran**
## Done

Multistage docker image.

## QA

- `export DOCKER_BUILDKIT=1`
- Build image with new dockerfile `docker build --tag myimage --build-arg BUILD_ID=myfakeid .`
- Run conainter with this image `docker run -ti -p 8045:80 myimage`
- Check build_id `myfakeid` is present `curl -i localhost:8045`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,48 @@
...
 
-# System dependencies
-RUN apt-get update && apt-get install -y --no-install-recommends python3 python3-setuptools python3-pip
+# Build the production image
+# ===
+FROM ubuntu:bionic
+
+# Install python and import python dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-lib2to3 python3-pkg-resources
+COPY --from=python-dependencies /root/.local/lib/python3.6/site-packages /root/.local/lib/python3.6/site-packages
```
Comment (carkod) [inline Dockerfile]: If you update to `ubuntu:focal` you might want to change this to python3.8.

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,48 @@
...
+# Build stage: Install python dependencies
+# ===
+FROM ubuntu:bionic AS python-dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-pip python3-setuptools
+ADD requirements.txt /tmp/requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install --user --requirement /tmp/requirements.txt
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:10-slim AS yarn-dependencies
```
[the remark this answers is not in the document]
Comment (jkfran) [reply Dockerfile]: Updated to 13-slim

File: Dockerfile [hunk 3]
```diff
@@ -1,20 +1,48 @@
...
+
+# Install python and import python dependencies
+RUN apt-get update && apt-get install --no-install-recommends --yes python3-lib2to3 python3-pkg-resources
+COPY --from=python-dependencies /root/.local/lib/python3.6/site-packages /root/.local/lib/python3.6/site-packages
+COPY --from=python-dependencies /root/.local/bin /root/.local/bin
+ENV PATH="/root/.local/bin:${PATH}"
+
+# Import code, build assets and mirror list
+ADD . .
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js
```
[the remark this answers is not in the document]
Comment (jkfran) [reply Dockerfile]: Thanks!

File: Dockerfile [hunk 4]
```diff
@@ -1,20 +1,48 @@
-FROM ubuntu:bionic
+# syntax=docker/dockerfile:experimental
+
+# Build stage: Install python dependencies
+# ===
+FROM ubuntu:bionic AS python-dependencies
```
[the remark this answers is not in the document]
Comment (jkfran) [reply Dockerfile]: Python 3.8 seems to not be compatible with gevent :/

**Pull request merged**